### ISSUE TITLE
Change in classpath ordering using Coursier

### ISSUE THREAD

**Issue by dwijnand** [sbt/sbt#5282]
## steps
sbt version: 1.3.4

## problem

(Apologies as I file this quickly, without a minimised reproduction.)

Both Play and Lagom implement a strategy of having alternative implementations for something (e.g. Akka HTTP vs Netty server backend in Play, akka-persistence-jdbc vs akka-peristence-jdbc in Lagom) by defining distinct jars with distinct classes, but which define the same config key:

```hocon
play.server {

  # The server provider class name
  provider = "play.core.server.NettyServerProvider"
```

```hocon
play {

  server {
    # The server provider class name
    provider = "play.core.server.AkkaHttpServerProvider"
```

```hocon
akka.persistence.journal.plugin = cassandra-journal
akka.persistence.snapshot-store.plugin = cassandra-snapshot-store
```

```hocon
akka.persistence.journal.plugin = jdbc-journal
akka.persistence.snapshot-store.plugin = jdbc-snapshot-store
```

For Play, the intent is that the Akka HTTP server is the default, and there are instructions to opt-out of that and opt-in to Netty.  For Lagom, I'm less sure, but I think there's no default and you just pick one _OR_ the other.

But after that strategy was set, Play's `Play-Test` module was made to depend on both, which means that the order of the jars defines which configuration wins.  Unfortunately, the ordering was accidentally correct in terms of our desires, meaning the config for Akka HTTP was seen, so we didn't become aware of this mistake.

The upgrade to sbt 1.3 (and, thus, from Ivy to Coursier) seems to switch the ordering which has revealed the mistake.

## expectation

Now, I say "mistake" because I don't think it's behaviour that should be relied upon.  A practice that is done, and should be possible, is to prepend jars to the classpath, the practice of "monkey-patching the classpath".  But I think the order of the classpath from managed dependency resolution shouldn't be relied upon.

So _I_ don't consider this a regression, but in speaking with @retronym, he advised that we should consider changing Coursier to maintain the previous behaviour, as it's still early days in sbt 1.3's adoption and we might be saving ourselves a lot of trouble down the line by changing this back for everyone.

## notes

* https://github.com/playframework/playframework/pull/9838 fixing our mistake
* https://github.com/playframework/playframework/pull/9877 adding complexity to our build by introducing asbt 1.3 as a testing variant... 😭
* https://github.com/lagom/lagom/pull/2523 fixing the same type of mistake, in Lagom

**Comment by eed3si9n** [comment]
I think the modules returned by Ivy are topologically sorted. Is that not the case with Coursier, or is it more subtle difference? If you want this to be addressed, I'd suggest coming up with a test case and submitting it to coursier/couriser.

**Comment by dwijnand** [comment]
Opened upstream as https://github.com/coursier/coursier/issues/1466.

**Comment by dwijnand** [comment]
@alexarchambault JFYI going to try to work on this, this week or next.

**Comment by dwijnand** [comment]
Fixed in #5421, coming in v1.3.9.
